lib/logflare_web/controllers/endpoints_controller.ex

```diff
@@ -16,14 +18,13 @@ defmodule LogflareWeb.EndpointsController do
     methods: ["GET", "POST", "OPTIONS"],
     send_preflight_response?: true
 
-  def query(conn, %{"token" => token}) do
-    endpoint_query = Endpoints.get_mapped_query_by_token(token)
-
-    case Endpoints.run_cached_query(endpoint_query, conn.query_params) do
-      {:ok, result} ->
-        Logger.debug("Endpoint cache result, #{inspect(result, pretty: true)}")
-        render(conn, "query.json", result: result.rows)
+  def query(%{assigns: %{endpoint: endpoint}} = conn, _params) do
+    endpoint_query = Endpoints.map_query_sources(endpoint)
 
+    with {:ok, result} <- Endpoints.run_cached_query(endpoint_query, conn.query_params) do
+      Logger.debug("Endpoint cache result, #{inspect(result, pretty: true)}")
+      render(conn, "query.json", result: result.rows)
+    else
       {:error, err} ->
         render(conn, "query.json", error: err)
```

Comment:
This could be removed since the fallback controller will know how to handle this

Reply:
lets do this change in a separate ticket & PR, currently we always return 200 for bigquery errors to bubble up, using fallback controller for this endpoint would result in status code change and api integration might go haywire.

Reply:
ticket [created](https://www.notion.so/supabase/Endpoints-api-controller-error-response-consistency-9b4d4272a75c48dd838de72c12f60cde?pvs=4)